### PR TITLE
Simplify Order messages form

### PR DESCRIPTION
**Pull request by JevgenijVisockij**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Simplifying order messages form
| Type?         | refacto
| Category?     | BO
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | Part of #16482
| How to test?  | Order messages form must work and look exactly the same

:notebook: **BC Break**
Backwards compatibility break introduced due to extension of TranslationAwareType instead of using TranslatorAwareTrait. This means if any module extends OrderMessage file they will get an exception.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20103)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by LouiseBonnard — DISMISSED** [review]

**Review by Junebyun — DISMISSED** [review]

**Review by matks — DISMISSED** [review]
Simple and efficient LGTM

**Review by PierreRambaud — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @JevgenijVisockij,

I find an issue following these steps:

1. Go to BO => Customer Service =>Order Messages
2. Create a new Order message 
3. Set a Name and the Message with invalid expressions 
4. Save
5. See error => there are no messages 

**With PR**
image

**Without PR**
image

Thanks!

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @JevgenijVisockij,

I found some issues:

1. If the Message is empty, we have a double alert and it is not well displayed
image

2. If the Message has invalid expressions, the alert is not displayed
image

3. If the Message is too long, the alert is not displayed
image


Thanks!

**Review by Progi1984 — DISMISSED** [review]

**Review by matks — DISMISSED** [review]
@JevgenijVisockij If the error management refactoring for TranslatableType is complete enough, LGTM

**Comment by matks** [comment]
Oh, little conflict 😓 blocking

**Comment by JevgenijVisockij** [comment]
This is no longer blocked by translatable errors and ready for reviews/Q&A

**Review by atomiix — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

File: src/PrestaShopBundle/Form/Admin/Sell/CustomerService/OrderMessageType.php [hunk 1]
```diff
@@ -30,15 +30,17 @@
...
 use PrestaShop\PrestaShop\Core\Domain\OrderMessage\OrderMessageConstraint;
 use PrestaShopBundle\Form\Admin\Type\TranslatableType;
-use Symfony\Component\Form\AbstractType;
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
 use Symfony\Component\Form\Extension\Core\Type\TextareaType;
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\Validator\Constraints\Length;
 
 /**
+ * Backwards compatibility break introduced in 1.7.8.0 due to extension of TranslationAwareType instead of using trait
```
Comment (eternoendless) [inline src/PrestaShopBundle/Form/Admin/Sell/CustomerService/OrderMessageType.php:39]: Not sure if this notice is needed here

**Review by matthieu-rolland — DISMISSED** [review]

**Comment by khouloudbelguith** [comment]
Hi @JevgenijVisockij,

I tried to run this test
`TEST_PATH="functional/BO/05_customerService/02_orderMessages/01_CRUDOrderMessage.js" URL_FO=http://prestashop.local/178x/ npm run specific-test`
But, I have errors detailed in this report
[mochawesome.html.zip](https://github.com/PrestaShop/PrestaShop/files/5608665/mochawesome.html.zip)
If this PR will be merged, the Nightly => NOK.
Ping @PrestaShop/qa-automation can you check this PR.

Thank you!

**Comment by JevgenijVisockij** [comment]
Thanks to @boubkerbribri testing should be fixed now, just need to restart Travis for this PR.

**Review by Progi1984 — APPROVED** [review]

File: src/PrestaShopBundle/Form/Admin/Sell/CustomerService/OrderMessageType.php (hunk 1 above)
Comment (atomiix) [reply src/PrestaShopBundle/Form/Admin/Sell/CustomerService/OrderMessageType.php:39]: ping @JevgenijVisockij :)

**Review by matks — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @JevgenijVisockij, @boubkerbribri,

It is ok :heavy_check_mark: 

Thank you!

**Comment by PierreRambaud** [comment]
Thank you @JevgenijVisockij

**Pull request merged**